### ISSUE TITLE
Explore global settings for cards

### ISSUE THREAD

**Issue by ravichdev** [material-components/material-design-for-wordpress#60]
<a href="https://github.com/jauyong"><img src="https://avatars.githubusercontent.com/u/5874851?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jauyong](https://github.com/jauyong)**
_Monday Dec 07, 2020 at 14:26 GMT_
_Originally opened as https://github.com/xwp/material-design-wp-plugin/issues/463_

----

## Feature description

Add options in the Customizer for global settings for cards (elevated and outlined).

This ticket was split from #362 
Note: there was a similar discussion regarding round corners.
Question: What about theme options that cover the same settings?
Note: Design may be needed here.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

1. Add toggle for all cards: Outlined or filled
2. Any card field should follow the guideline set above, unless a user has specifically chosen something at the block level. 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by ravichdev** [comment]
<a href="https://github.com/jwold"><img src="https://avatars.githubusercontent.com/u/3411173?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jwold](https://github.com/jwold)**
_Thursday Jan 14, 2021 at 19:20 GMT_

----

Waiting on feedback from: https://github.com/xwp/material-design-wp-plugin/issues/462#issuecomment-760413320

**Comment by ravichdev** [comment]
<a href="https://github.com/jauyong"><img src="https://avatars.githubusercontent.com/u/5874851?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jauyong](https://github.com/jauyong)**
_Friday Feb 05, 2021 at 16:52 GMT_

----

@jwold #462 has been updated

**Comment by ravichdev** [comment]
<a href="https://github.com/jauyong"><img src="https://avatars.githubusercontent.com/u/5874851?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jauyong](https://github.com/jauyong)**
_Friday Feb 12, 2021 at 16:34 GMT_

----

@dawidmlynarz to look into this as well. 

Need to consider how toggling in the customizer affects the toggle at the block level. 

Note, see:
![image](https://user-images.githubusercontent.com/5874851/107795204-5fe59b80-6d26-11eb-8788-5bb1cfe2135d.png)

**Comment by ravichdev** [comment]
<a href="https://github.com/dawidmlynarz"><img src="https://avatars.githubusercontent.com/u/8026823?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dawidmlynarz](https://github.com/dawidmlynarz)**
_Wednesday Feb 17, 2021 at 10:41 GMT_

----

@jauyong The design applies also to #462 since it solves the same problem. Below you can see my solution proposal:
<img width="1290" alt="Global Styles" src="https://user-images.githubusercontent.com/8026823/108192533-a9fac280-7114-11eb-8a48-e525919169f5.png">

I'd suggest having a look at other settings that might have a global effect on styles and group them inside of this section. This would increase the discoverability of the same type of settings.

Looking forward to your feedback.

**Comment by ravichdev** [comment]
<a href="https://github.com/jauyong"><img src="https://avatars.githubusercontent.com/u/5874851?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jauyong](https://github.com/jauyong)**
_Friday Feb 19, 2021 at 18:02 GMT_

----

@jauyong to update messaging. 
and labels to ( Card, Text).

**Comment by ravichdev** [comment]
<a href="https://github.com/jauyong"><img src="https://avatars.githubusercontent.com/u/5874851?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jauyong](https://github.com/jauyong)**
_Friday Feb 19, 2021 at 18:12 GMT_

----

@dawidmlynarz 
Can you update the field labels in the mocks by removing the word "blocks".
Card Blocks --> Cards
Text Blocks --> Text

We also need to update the messaging displayed. 
The way this will work is that updating the global style will update any existing block that is currently using the "default"/global style. ie If the local setting was changed from the default/global, updating the global will not override the style. Any new blocks will now inherit the global styling by default.

**Comment by ravichdev** [comment]
<a href="https://github.com/dawidmlynarz"><img src="https://avatars.githubusercontent.com/u/8026823?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dawidmlynarz](https://github.com/dawidmlynarz)**
_Tuesday Feb 23, 2021 at 07:50 GMT_

----

Thank you for the feedback, I have changed the labels:
Card Blocks --> Cards
Text Blocks --> Text **field** (this is styling the input fields to just 'Text' is too general and not that self-explanatory)

Below you will find a revised version of the messaging. Let me know if you have any suggestions to improve the wording.

<img width="310" alt="Screenshot 2021-02-23 at 08 46 32" src="https://user-images.githubusercontent.com/8026823/108814770-abfccf80-75b3-11eb-84a0-bf0237efc9c4.png">

**Comment by PatelUtkarsh** [comment]
@jauyong I've added a comment [here](https://github.com/material-components/material-design-for-wordpress/pull/127#issuecomment-842340027) about the dynamic vs static block, and it's effect based on settings of this issue - 

> Global styles will be applied to any new and existing static block using the default style. This will not affect any local style overrides. 

Can you check if above message works as description? cc: @dawidmlynarz

**Comment by PatelUtkarsh** [comment]
I was thinking about avoiding confusion and make this more clear:

Maybe we can introduce either of the following:

1. Force override boolean customizer setting for card and text field - which will change card / text-field based on global setting using JavaScript.

2. Or we can introduce off default state in current settings and assume force when it's not off. (And update description accordingly).

@jauyong any thoughts?

**Comment by csossi** [comment]
Reading the thread above and just want to ensure we have the right copy here:
![image](https://user-images.githubusercontent.com/16945664/125211997-541be800-e278-11eb-8125-0fbf3c2f4175.png)

This all right? Should it be "Global styles will be applied to any new and existing static block using the default style. This will not affect any local style overrides for supported blocks."

**Comment by PatelUtkarsh** [comment]
I think this is a correct copy, We have changed the approach and now we are applying settings via JS on the client side which is more "global" instead of having it condition on static vs dynamic blog. @jauyong Thoughts?

**Comment by jauyong** [comment]
@PatelUtkarsh are you saying that even if a user sets local style overrides at the block level, any changes made at the global level will override those settings?

**Comment by PatelUtkarsh** [comment]
@jauyong Yes, the global style is acting like force override all, unless it's set as inherit in that case it won't.

**Comment by jauyong** [comment]
@pbearne can you please document the proposed solution here. What settings will be displayed on the global and on the block level views?

**Comment by jauyong** [comment]
I have mapped out the states and change states here: https://miro.com/app/board/o9J_l5og-7s=/

**Comment by csossi** [comment]
@PatelUtkarsh , does the AC need updating here? Also, the Implementation notes "Force" options - has this changed? Any QA Instructions you can share would be most helpful
